pilot/pkg/model/sidecar.go

```diff
@@ -447,10 +448,11 @@ func convertIstioListenerToWrapper(ps *PushContext, configNamespace string,
 			out.listenerHosts[parts[0]] = make([]host.Name, 0)
 		}
 		out.listenerHosts[parts[0]] = append(out.listenerHosts[parts[0]], host.Name(parts[1]))
+		listenerHosts = append(listenerHosts, host.Name(parts[1]))
 	}
 
 	vses := ps.VirtualServicesForGateway(configNamespace, constants.IstioMeshGateway)
-	out.virtualServices = SelectVirtualServices(vses, out.listenerHosts)
+	out.virtualServices = SelectVirtualServices(vses, listenerHosts)
```

Comment:
Its not obvious why we use different listenerHosts. I sort of get it in the context of this PR, but the code would be confusing looking at it later. Can we add comments?